Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -0,0 +1,551 @@
+
+  -- `γ` and `γ'` when expressed in the local chart are continuous on this neighbourhood
+  have hcont := (continuousOn_extChartAt I (γ t₀)).comp
+    (IsIntegralCurveOn.continuousOn hmfd) hsrc
+  have hcont' := (continuousOn_extChartAt I (γ' t₀)).comp
+    (IsIntegralCurveOn.continuousOn hmfd') hsrc'
+
+  simp_rw [Real.ball_eq_Ioo] at hmem hsrc hmfd hcont hmem' hsrc' hmfd' hcont'
+
+  -- `γ` and `γ'` are
+  have heqon : EqOn ((extChartAt I (γ t₀)) ∘ γ) ((extChartAt I (γ' t₀)) ∘ γ')
+    (Ioo (t₀ - ε) (t₀ + ε)) := by
+    -- uniqueness of ODE solutions in an open interval
+    apply ODE_solution_unique_of_mem_set_Ioo hlip (t₀ := t₀)
+      (Real.ball_eq_Ioo _ _ ▸ (Metric.mem_ball_self hε)) hcont _ hmem hcont' _ hmem' (by simp [h])
+    · intros t ht
+      rw [hv']
+      have := hmfd.hasDerivAt ht (hsrc t ht)
+      apply this.congr_deriv
+      have : γ t = (extChartAt I (γ t₀)).symm (((extChartAt I (γ t₀)) ∘ γ) t) := by
+        rw [Function.comp_apply, PartialEquiv.left_inv]
+        exact hsrc t ht
+      rw [this]
+    · intros t ht
+      rw [hv', h]
+      have := hmfd'.hasDerivAt ht (hsrc' t ht)
+      apply this.congr_deriv
+      have : γ' t = (extChartAt I (γ' t₀)).symm (((extChartAt I (γ' t₀)) ∘ γ') t) := by
+        rw [Function.comp_apply, PartialEquiv.left_inv]
+        exact hsrc' t ht
+      rw [this]
+
+  -- finally show `EqOn γ γ' _` by composing with the inverse of the local chart around `γ t₀`
+  refine EqOn.trans ?_ (EqOn.trans (heqon.comp_left (g := (extChartAt I (γ t₀)).symm)) ?_)
+  · intros t ht
+    rw [Function.comp_apply, Function.comp_apply, PartialEquiv.left_inv _ (hsrc _ ht)]
+  · intros t ht
+    rw [Function.comp_apply, Function.comp_apply, h, PartialEquiv.left_inv _ (hsrc' _ ht)]
+
+theorem isIntegralCurveAt_eqOn_of_contMDiffAt_boundaryless [BoundarylessManifold I M]
+    (hv : ContMDiffAt I I.tangent 1 (fun x => (⟨x, v x⟩ : TangentBundle I M)) (γ t₀))
+    (hγ : IsIntegralCurveAt γ v t₀) (hγ' : IsIntegralCurveAt γ' v t₀) (h : γ t₀ = γ' t₀) :
+    ∃ ε > 0, EqOn γ γ' (Ioo (t₀ - ε) (t₀ + ε)) :=
+  isIntegralCurveAt_eqOn_of_contMDiffAt (BoundarylessManifold.isInteriorPoint I) hv hγ hγ' h
+
+variable [T2Space M] {a b : ℝ}
+
+/-- Integral curves are unique on open intervals.
+
+  If a continuously differentiable vector field `v` admits two integral curves `γ γ' : ℝ → M`
+  on some open interval `Ioo a b`, and `γ t₀ = γ' t₀` for some `t ∈ Ioo a b`, then `γ` and `γ'`
+  agree on `Ioo a b`. -/
+theorem isIntegralCurveOn_Ioo_eqOn_of_contMDiff (ht₀ : t₀ ∈ Ioo a b)
+    (hγt : ∀ t ∈ Ioo a b, I.IsInteriorPoint (γ t))
+    (hv : ContMDiff I I.tangent 1 (fun x => (⟨x, v x⟩ : TangentBundle I M)))
+    (hγ : IsIntegralCurveOn γ v (Ioo a b)) (hγ' : IsIntegralCurveOn γ' v (Ioo a b))
+    (h : γ t₀ = γ' t₀) : EqOn γ γ' (Ioo a b) := by
+  /-
+  strategy:
+  * Lee P.213, just need to translate "S is closed in J" to type theory language
+  -/
+  set s := {t | γ t = γ' t} ∩ Ioo a b with hs
+  have hsub : Ioo a b ⊆ s := by
```

Comment:
What do you think about using `suffices` here?
`  suffices h : Ioo a b ⊆ s from fun t ht ↦ mem_setOf.mp ((subset_def ▸ h) t ht).1`,
and then `hsub` becomes your main goal.